models/ed/R/SDA.helpers.ED2.R

```diff
@@ -0,0 +1,63 @@
+#' @title Get ED history restart file path
+#'
+#' @author Alexey Shiklomanov
+#' @param mod_outdir Directory where PEcAn stores ensemble outputs. Usually 
+#' \code{<workflowID>/out}
+#' @param runid PEcAn run ID
+#' @param file.time Start or end time from SDA analysis
+get_restartfile.ED2 <- function(mod_outdir, runid, file.time) {
+    runid <- as.character(runid)
+    histfile_path <- file.path(mod_outdir, runid)
+
+    file_year <- lubridate::year(file.time)
+    file_month <- lubridate::month(file.time)
+    file_day <- lubridate::day(file.time)
+    datetime_string <- sprintf("%04d-%02d-%02d-000000", 
+                               file_year, 
+                               file_month,
+                               file_day)
+    histfile_string <- paste0("history-S-",
+                              datetime_string,
+                              ".*\\.h5$")
+
+    histfile <- list.files(histfile_path, 
+                           histfile_string, 
+                           full.names = TRUE)
```

Comment:
Including the paths. By default, `list.files` (which is a base R function), only lists the file names, regardless of where you point it. `full.names` lists the full directory path, which is unambiguous.